controllers/azurecluster_reconciler_test.go

```diff
@@ -246,10 +334,23 @@ func TestAzureClusterServiceDelete(t *testing.T) {
 			svcThreeMock := mock_azure.NewMockServiceReconciler(mockCtrl)
 
 			tc.expect(groupsMock.EXPECT(), vnetpeeringsMock.EXPECT(), svcOneMock.EXPECT(), svcTwoMock.EXPECT(), svcThreeMock.EXPECT())
+			c := tc.clientBuilder(g)
 
 			s := &azureClusterService{
 				scope: &scope.ClusterScope{
-					AzureCluster: &infrav1.AzureCluster{},
+					Client: c,
+					AzureCluster: &infrav1.AzureCluster{
+						Spec: infrav1.AzureClusterSpec{
+							ResourceGroup: resourceGroup,
+						},
+					},
+					Cluster: &clusterv1.Cluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:              clusterName,
+							Namespace:         namespace,
+							DeletionTimestamp: &metav1.Time{Time: time.Now()},
+						},
+					},
```

Comment:
This may be out of scope for this PR but it could be worth adding some test cases for `UseLegacyGroups: true` to cover all cases.

Reply:
I did try adding some but it turned out to be super messy and I couldn't justify fighting through that since ideally we'll drop that flag entirely in the next week or so. I can come back to these if we think it will stick around much longer than that.